frontend/middleware_validatestatic.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"errors"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/go-chi/chi/v5"
```

Comment:
we should avoid bringing `chi` in unless we have to. To get URLParams, we can use built in go `request.PathValue()` This may require us to break the patterns we currently have in `frontend.go` along with `const.go` to break out the "resourceProvider"

Reply:
The `http.ServeMux` routes are set up with the resource provider namespace hard-coded into the path.  So if the namespace doesn't match, the request will 404 and not even reach this middleware.  For the resource type I agree with your comment below.